### PR TITLE
Implement get user by id API

### PR DESCRIPTION
**Pull request by Bogomazz**

### CODE REVIEW

File: HdProduction.HelpDesk.Infrastructure/Services/UserService.cs [hunk 1]
```diff
@@ -14,12 +15,20 @@ public UserService(IUserRepository userRepository)
...
-        public Task CreateAsync(string email, string pwdHash)
+        public async Task<User> CreateAsync(string firstName, string lastName, string email, string pwdHash)
         {
             var pwdHelper = SecurityHelper.Create();
-            _userRepository.Add(new User(email, "abc", "cde", "", 
-                pwdHelper.CreateSaltedPassword(pwdHash), pwdHelper.Salt));
-            return _userRepository.SaveAsync();
+            var user = new User(email, firstName, lastName, "",
+                pwdHelper.CreateSaltedPassword(pwdHash), pwdHelper.Salt);
+            _userRepository.Add(user);
```
Comment (farexy) [inline HdProduction.HelpDesk.Infrastructure/Services/UserService.cs:28]: add class UserSafeguard : IUserSafeguard
with validations like

void EnsureEmail(string email){
  if(regex.NotMatch(email)){
    throw new BusinessLogic("Not valid email");
  }
  if (_userRepository.FindByEmail(email) != null){
     throw new BusinessLogic("Already exist");
  }
}

File: HdProduction.HelpDesk.Infrastructure/Services/UserService.cs [hunk 2]
```diff
@@ -14,12 +15,20 @@ public UserService(IUserRepository userRepository)
...
+            await _userRepository.SaveAsync();
+            return user;
+        }
+
+        public async Task<User> Find(long id)
```
Comment (farexy) [inline HdProduction.HelpDesk.Infrastructure/Services/UserService.cs]: FindAsync
by convention write Async postfix everywhere in asynchronous methods that you call from code
(controller methods is an exception)

File: HdProduction.HelpDesk.Api/Controllers/UsersController.cs [hunk 3]
```diff
@@ -10,16 +12,31 @@ namespace HdProduction.HelpDesk.Api.Controllers
...
+        public async Task<UserResponseModel> Create(UserRequestModel requestModel)
         {
-            return _userService.CreateAsync(requestModel.Email, requestModel.PwdHash);
+            var user = await _userService.CreateAsync(
+                requestModel.FirstName,
+                requestModel.LastName,
+                requestModel.Email,
+                requestModel.PwdHash
+            );
+            return _mapper.Map<User, UserResponseModel>(user);
```
Comment (farexy) [inline HdProduction.HelpDesk.Api/Controllers/UsersController.cs]: we usually do not return object from service on create, update etc.
prefer return id from Create method and then call Find(id)
ef caches entity inside scope, so there won't be any overhead

File: HdProduction.HelpDesk.Infrastructure/Services/UserService.cs (hunk 1 above)
Comment (Bogomazz) [reply HdProduction.HelpDesk.Infrastructure/Services/UserService.cs:28]: I would like to see it in your example:)

Comment (farexy) [reply HdProduction.HelpDesk.Infrastructure/Services/UserService.cs:28]: added commit with example

File: HdProduction.HelpDesk.Api/Controllers/UsersController.cs [hunk 4]
```diff
@@ -10,16 +12,32 @@ namespace HdProduction.HelpDesk.Api.Controllers
...
         {
-            return _userService.CreateAsync(requestModel.Email, requestModel.PwdHash);
+            var userId = await _userService.CreateAsync(
+                requestModel.FirstName,
+                requestModel.LastName,
+                requestModel.Email,
+                requestModel.PwdHash
+            );
+            var user = await _userService.FindAsync(userId);
+            return _mapper.Map<User, UserResponseModel>(user);
```
Comment (farexy) [inline HdProduction.HelpDesk.Api/Controllers/UsersController.cs]: just call Find method of controller:)

File: HdProduction.HelpDesk.Infrastructure/Services/UserService.cs (hunk 1 above)
Comment (Bogomazz) [reply HdProduction.HelpDesk.Infrastructure/Services/UserService.cs:28]: Let keep it as is for now. But actually, I don't like this approach. But let discuss it later.

File: HdProduction.HelpDesk.Api/Controllers/UsersController.cs [hunk 5]
```diff
@@ -1,25 +1,54 @@
...
+        [HttpGet("{id}")]
+        public async Task<UserResponseModel> Find(long id)
+        {
+            var user = await _userService.FindAsync(id);
+            return _mapper.Map<User, UserResponseModel>(user);
+        }
+
+        [HttpGet("me")]    
+        [Authorize(AuthenticationSchemes = JwtDefaults.AuthenticationSchemeIgnoreExpiration)]
+
```
Comment (farexy) [inline HdProduction.HelpDesk.Api/Controllers/UsersController.cs]: why did you use here ignore expiration auth scheme?

P.S. : what an awful line break.. fix it pls!

Comment (Bogomazz) [reply HdProduction.HelpDesk.Api/Controllers/UsersController.cs]: Cause I tried to find working approach:)

**Pull request merged**